code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/Optimize.java

```diff
@@ -118,6 +118,136 @@ public static void updatePropositions(
+     * @param callback {@code AdobeCallback<Map<DecisionScope, OptimizeProposition>>} which will be
+     *     invoked when decision propositions are received from the Edge network.
+     */
+    public static void updatePropositions(
+            @NonNull final List<DecisionScope> decisionScopes,
+            @Nullable final Map<String, Object> xdm,
+            @Nullable final Map<String, Object> data,
+            @NonNull final AdobeCallback<Map<DecisionScope, OptimizeProposition>> callback) {
+        if (OptimizeUtils.isNullOrEmpty(decisionScopes)) {
+            Log.warning(
+                    OptimizeConstants.LOG_TAG,
+                    SELF_TAG,
+                    "Cannot update propositions, provided list of decision scopes is null or"
+                            + " empty.");
+            AEPOptimizeError aepOptimizeError =
+                    new AEPOptimizeError(null, null, null, null, AdobeError.UNEXPECTED_ERROR);
+            failWithOptimizeError(callback, aepOptimizeError);
+            return;
+        }
+
+        final List<DecisionScope> validScopes = new ArrayList<>();
+        for (final DecisionScope scope : decisionScopes) {
+            if (!scope.isValid()) {
+                continue;
+            }
+            validScopes.add(scope);
+        }
+
+        if (validScopes.size() == 0) {
+            Log.warning(
+                    OptimizeConstants.LOG_TAG,
+                    SELF_TAG,
+                    "Cannot update propositions, provided list of decision scopes has no valid"
+                            + " scope.");
+            return;
+        }
+
+        final List<Map<String, Object>> flattenedDecisionScopes = new ArrayList<>();
+        for (final DecisionScope scope : validScopes) {
+            flattenedDecisionScopes.add(scope.toEventData());
+        }
+
+        final Map<String, Object> eventData = new HashMap<>();
+        eventData.put(
+                OptimizeConstants.EventDataKeys.REQUEST_TYPE,
+                OptimizeConstants.EventDataValues.REQUEST_TYPE_UPDATE);
+        eventData.put(OptimizeConstants.EventDataKeys.DECISION_SCOPES, flattenedDecisionScopes);
+
+        if (!OptimizeUtils.isNullOrEmpty(xdm)) {
+            eventData.put(OptimizeConstants.EventDataKeys.XDM, xdm);
+        }
+
+        if (!OptimizeUtils.isNullOrEmpty(data)) {
+            eventData.put(OptimizeConstants.EventDataKeys.DATA, data);
+        }
+
+        final Event event =
+                new Event.Builder(
+                                OptimizeConstants.EventNames.UPDATE_PROPOSITIONS_REQUEST,
+                                OptimizeConstants.EventType.OPTIMIZE,
+                                OptimizeConstants.EventSource.REQUEST_CONTENT)
+                        .setEventData(eventData)
+                        .build();
+
+        MobileCore.dispatchEventWithResponseCallback(
+                event,
+                OptimizeConstants.GET_RESPONSE_CALLBACK_TIMEOUT,
```

Comment:
please use "EDGE_CONTENT_COMPLETE_RESPONSE_TIMEOUT" here instead of "GET_RESPONSE_CALLBACK_TIMEOUT".

Reply:
done